mesonbuild/mesonmain.py

```diff
@@ -62,21 +62,31 @@ def setup_vsenv():
     if shutil.which('cl.exe'):
         return
 
-    bat_locator_bin = r'c:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe'
-    if not os.path.exists(bat_locator_bin):
+    root = os.environ.get("ProgramFiles(x86)") or os.environ.get("ProgramFiles")
+    bat_locator_bin = pathlib.Path(root, 'Microsoft Visual Studio/Installer/vswhere.exe')
+    if not bat_locator_bin.exists():
         return
-    bat_json = subprocess.check_output([bat_locator_bin, '-latest', '-format', 'json'])
+    bat_json = subprocess.check_output(
+        [
+            str(bat_locator_bin),
+            '-latest',
+            '-prerelease',
+            '-requiresAny',
+            '-requires', 'Microsoft.VisualStudio.Component.VC.Tools.x86.x64',
+            '-requires', 'Microsoft.VisualStudio.Workload.WDExpress',
```

Comment:
I think we only require `Microsoft.VisualStudio.Component.VC.Tools.x86.x64`, what's `Microsoft.VisualStudio.Workload.WDExpress` ?

If I understand correctly, with `-requiresAny` it will accept any of those 2, so that probably still works even if I don't have WDExpress installed, but if I have WDExpress installed but not Tools.x86.x64 it will work but won't have the C++ toolchain. If I read the [doc](https://docs.microsoft.com/en-us/visualstudio/install/workload-component-id-vs-express?view=vs-2017) correctly WDExpress only contains ARM and ARM64 toolchains, really weird...

Disclaimer: I'm not a VisualStudio expert by any mean, I could be totally wrong :D

Reply:
I think I should remove `Microsoft.VisualStudio.Workload.WDExpress`.

> If I understand correctly, with -requiresAny it will accept any of those 2, so that probably still works even if I don't have WDExpress installed, but if I have WDExpress installed but not Tools.x86.x64 it will work but won't have the C++ toolchain.

I think this is how it is going to work.